### PR TITLE
Add dr_util to make uploads to DR console easier

### PR DESCRIPTION
**Pull request by breadcentric**
The main addition is a dr_util script which helps with manipulating the models. Two key features are:
* preparing model snapshots with the latest Step data from a selected model - such snapshot has only five files and can be used as a submission model to a virtual race or as a pretrained model
* uploading model snapshots to DeepRacer bucket - it assumes a single model is used for all local submissions, but with use of label when creating a snapshot, the label is written to evaluation logs which makes it easier to detect which model was submitted for evaluation

I have also updated readme with my findings about providing hyperparameters and setting nvidia runtime in docker-compose with minimal update to `rl_deepracer_coach_robomaker.py`

### CODE REVIEW

File: dr_util.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+def init_config(config_file=default_config_file):
+    if os.path.exists(config_file):
+        logging.warning("%s exists, doing nothing", config_file)
+        return
+
+    import json
+    with open(config_file, 'w') as f:
+        empty_config = {
+            'minio_bucket_base': '???',
```
Comment (crr0004) [inline dr_util.py]: Can we change this to either sane defaults or with an example value? It's not clear what they should be from the placeholders

File: dr_util.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                        "config first running command init, see --help",
+                        config_file)
+        return False
+    import json
+
+    global config
+    with open(config_file) as f:
+        config = json.load(f)
+
+        config['training_path'] = config['minio_bucket_base'] + '/rl-deepracer-sagemaker'
```
Comment (crr0004) [inline dr_util.py]: These should be able to be overwritten in the config, possibly pulled from the same environment variables. The paths aren't always what I've set them to, they're just defaults

File: README.md [hunk 3]
```diff
@@ -94,6 +96,24 @@ You now specify your action space in the json file you pass in through
 `MODEL_METADATA_FILE_S3_KEY`, which is defaulted to
 `bucket/custom_files/model_metadata.json`
 
+### Altering hyperparameters
```
Comment (crr0004) [inline README.md]: This part is in the wiki. I don't want everything to be in the README because it gets too long and very difficult to read. Can you submit a different PR against the wiki with this content?

File: README.md [hunk 4]
```diff
@@ -94,6 +96,24 @@ You now specify your action space in the json file you pass in through
...
+To override default hyperparameters, in `rl_deepracer_coach_robomaker.py` uncomment the fields that you wish to override and set their value.
+
+Default values can be found in `sagemaker_graph_manager.py`
+
+### Using a pretrained model
```
Comment (crr0004) [inline README.md]: This has a part in the wiki as well. A PR needs to be there

File: .gitignore [hunk 5]
```diff
@@ -104,3 +104,11 @@ venv.bak/
 .mypy_cache/
 
 *.tar.gz
+
+# just some dev files
+data
```
Comment (crr0004) [inline .gitignore]: These should be in a global ignore for your environment. The dr_util_config.json can stay though

File: README.md [hunk 6]
```diff
@@ -94,6 +96,24 @@ You now specify your action space in the json file you pass in through
...
+To train based on an existing model, copy it into `rl-deepracer-pretrained` in bucket and uncomment options
+`pretrained_s3_bucket` and `pretrained_s3_prefix` in `rl_deepracer_coach_robomaker.py`
+
+### dr_util.py
```
Comment (crr0004) [inline README.md:99]: Can you add a warning of not taking any responsibility of what this does to AWS accounts? I know that goes for anything you don't write yourself, but because this is directly interacting with an AWS account, I think it needs an extra warning.

**Review by crr0004 — CHANGES_REQUESTED** [review]
The util file looks very promising and very helpful. Thank you a lot for doing this. I have added some comments on on the relevant lines. They might come across as aggressive, but I just wanted to get to the point in the line comments. I am very happy to get this into the repo and most of comments about are ensuring we don't make assumptions about people's setup

File: README.md (hunk 3 above)
Comment (breadcentric) [reply README.md]: I raised #29 - I cannot submit PRs against the wiki

File: README.md (hunk 4 above)
Comment (breadcentric) [reply README.md]: I raised #29 - I cannot submit PRs against the wiki

File: README.md (hunk 6 above)
Comment (breadcentric) [reply README.md:99]: Added here and in script

File: dr_util.py (hunk 1 above)
Comment (breadcentric) [reply dr_util.py]: I added values describing what they should be. Hope you like it this way.

File: dr_util.py (hunk 2 above)
Comment (breadcentric) [reply dr_util.py]: Done

File: .gitignore (hunk 5 above)
Comment (breadcentric) [reply .gitignore]: Done

**Comment by breadcentric** [comment]
I applied changes, hope it's OK now.

File: README.md (hunk 6 above)
Comment (crr0004) [reply README.md:99]: Looks good, cheers!

File: dr_util.py (hunk 1 above)
Comment (crr0004) [reply dr_util.py]: Looks good, cheers!

File: dr_util.py (hunk 2 above)
Comment (crr0004) [reply dr_util.py]: Looks good, cheers!

File: README.md (hunk 3 above)
Comment (crr0004) [reply README.md]: Ah okay. I didn't realise you can't raise PRs against the wiki. I assumed because it's actually a repo underneath you could, but you are right.

**Review by crr0004 — APPROVED** [review]
